python/mspasspy/db/database.py

```diff
@@ -1519,7 +1519,7 @@ def _check_required(self, collection='site', keys=['lat','lon','elev','starttime
         return tuple([wrong_types,undef])
 
     def update_metadata(self, mspass_object, mode='promiscuous', exclude_keys=None, collection=None, ignore_metadata_changed_test=False,
-                        data_tag=None, alg_name='update_data', alg_id='0'):
+                        data_tag=None, alg_name='update_data', alg_id='0', is_save_data=False):
```

Comment:
Well... because this method is public to the users, I don't think it is a good idea to include options that are for internal use mostly. Well, since we are at this now, I guess it may not be a bad idea to change the structure a little bit and create a private `_update_metadata` method that is called by the `update_metadata` and `save_data` methods. This way, the code should look cleaner, too. What do you think?

Reply:
Yeah, I had doubts when I implemented this method, now it seems like what you suggest is a good idea.